src/components/panel/panel.js

```diff
@@ -832,6 +850,47 @@ MdPanelRef.prototype._addStyles = function() {
 };
 
 
+/**
+ * Add ARIA-specific attributes appropriate for panels.
+ */
+MdPanelRef.prototype._configureAria = function() {
+  this._panelEl.attr('tabIndex', '-1');
+
+  this._configureTrapFocus();
+};
+
+
+/**
+ * Setup the focus traps.
+ * @private
+ */
+MdPanelRef.prototype._configureTrapFocus = function() {
+  if (this._config.trapFocus) {
+    var element = this._panelEl;
+    // Set up elements before and after the panel to capture focus and
+    // redirect back into the panel.
+    this._topFocusTrap = document.createElement('div');
+    this._topFocusTrap.classList.add('_md-panel-focus-trap');
+    this._topFocusTrap.tabIndex = 0;
```

Comment:
I'd create one constant `HTMLElement` like `FOCUS_TRAP_TEMPLATE` with these properties already set and then clone it for both the top and bottom.